### ISSUE TITLE
unmatched-braces and indent autofixes can conflict when applied together, producing malformed output

### ISSUE THREAD

**Issue by walf443** [walf443/nginx-lint#296]
## Summary

Discovered while fixing #294 (`find_close_brace_offset` now correctly anchors a missing closing brace to a marker comment instead of always falling back to EOF). With that fix in place, running `nginx-lint --fix` with the **default rule set** (not `--rule-only unmatched-braces`) on `tests/fixtures/rules/syntax/unmatched_braces/001_basic/error/nginx.conf` no longer matches the fixture's `expected/nginx.conf`, even though `unmatched-braces`'s own fix is correct in isolation.

## Repro

```
$ nginx-lint --rule-only unmatched-braces --fix file.conf   # correct
$ nginx-lint --rule-only unmatched-braces,indent --fix file.conf   # malformed
```

With just `unmatched-braces` alone, the three missing closing braces land exactly where expected. Adding `indent` into the same `--fix` run (indent also proposes fixes for the same fixture, since several lines are marked as having wrong indentation) produces a mangled result — for the `http`-level brace specifically, the applied text is `      }` (6 spaces of indentation, matching neither rule's own indent) immediately followed by trailing whitespace stuck onto the marker comment's line (`# Missing closing brace for http      `), rather than the clean `}` / comment on its own line that either rule alone would produce.

## Why this wasn't caught by `test_all_rule_fixtures`

That test filters to only the target rule's own fixes before calling `apply_fixes` (`all_errors.iter().filter(|e| e.rule == tc.rule_name && !e.fixes.is_empty())`), so it never exercises multiple rules' fixes being applied in the same batch. This is the right scope for that test (verifying each rule's fix correctness in isolation), but it means fix *interaction* bugs across rules are invisible to it.

## Suspected root cause

`apply_fixes_to_content_detailed` (`crates/nginx-lint-common/src/linter.rs`) only guards against literal byte-range overlap between fixes before applying them (sorted descending by offset, applied via sequential `String::replace_range` calls). It has no notion of one fix's insertion semantically invalidating another's — e.g. `unmatched-braces` inserting a `}` immediately before a line that `indent` is *also* rewriting the leading whitespace of. Both fixes' offsets are computed against the original (pre-fix) content, and applying them sequentially without re-validating against each other's effects can produce output neither rule intended.

This is a general risk for any two rules whose fixes touch adjacent/nearby regions of the same file, not specific to unmatched-braces + indent — those two just happen to be the first observed combination.

## Suggested next step

Investigate whether `apply_fixes_to_content_detailed` needs a stronger conflict model (e.g. also skip a fix if it starts exactly at another already-applied insertion's position but produces a result inconsistent with either fix's own intent), or whether individual rules should avoid proposing fixes anchored to the same line another active rule is also rewriting.
